File: 338-CountingBits/338-CountingBits.py
# Last updated: 4/29/2026, 10:31:11 AM
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = []
        def count(i):
            count = 0
            num = i
            while(num > 0):
                remainder = num % 2
                if remainder == 1:
                    count += 1
                num = (num - remainder) / 2
            return count
        for i in range(0, n+1):
            res.append(count(i))
        
        return res

File: 338-CountingBits/338-CountingBits.js
// Last updated: 4/29/2026, 10:31:06 AM
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    const countOnes = num => {
        let i = num;
        let count = 0;
        while(i > 0) {
            if(i % 2 === 1) {
                count++;
                i--;
            }
            i /= 2;
        }
        return count;
    }
    const res = [];
    for(let i = 0; i <= n; i++) {
        res.push(countOnes(i));
    }

    return res;
};